src/integer/poly_over_z/coefficient_embedding.rs

```diff
@@ -0,0 +1,150 @@
+        SetCoefficient, SetEntry,
+    },
+};
+
+impl IntoCoefficientEmbedding<MatZ> for PolyOverZ {
+    /// Computes the coefficient embedding of the polynomial
+    /// in a [`MatZ`] as a column vector, where the i-th entry
+    /// of the vector corresponds to the i-th coefficient.
+    ///
+    /// Parameters:
+    /// - `size`: determines the length of the vector
+    ///
+    /// Returns a coefficient embedding as a vector
+    ///
+    /// # Examples
+    /// ```
+    /// use std::str::FromStr;
+    /// use qfall_math::{
+    ///     integer::{MatZ, PolyOverZ},
+    ///     traits::IntoCoefficientEmbedding,
+    /// };
+    ///
+    /// let poly = PolyOverZ::from_str("3  17 3 -5").unwrap();
+    /// let vector = poly.into_coefficient_embedding(4);
+    /// let cmp_vector = MatZ::from_str("[[17],[3],[-5],[0]]").unwrap();
+    /// assert_eq!(cmp_vector, vector)
+    /// ```
+    ///
+    /// # Panics ...
+    /// - if `size` is not larger than the degree of the polynomial, i.e.
+    /// the coefficients can not be embedded correctly.
+    fn into_coefficient_embedding(&self, size: impl Into<i64>) -> MatZ {
+        let size = size.into();
+        assert!(size > self.get_degree());
+        let mut out = MatZ::new(size, 1);
+        for j in 0..size {
+            match self.get_coeff(j) {
+                Ok(value) => out.set_entry(j, 0, &value).unwrap(),
+                Err(_) => break,
+            }
+        }
+        out
+    }
+}
+
+impl FromCoefficientEmbedding<&MatZ> for PolyOverZ {
+    /// Computes a polynomial from a vector.
+    /// The first i-th entry of the column vector is taken
+    /// as the coefficient of the polynomial.
+    ///
+    /// Parameters:
+    /// - `embedding`: the column vector that encodes the embedding
+    ///
+    /// Returns a polynomial that corresponds to the embedding
+    ///
+    /// # Examples
+    /// ```
+    /// use std::str::FromStr;
+    /// use qfall_math::{
+    ///     integer::{MatZ, PolyOverZ},
+    ///     traits::FromCoefficientEmbedding,
+    /// };
+    ///
+    /// let vector = MatZ::from_str("[[17],[3],[-5]]").unwrap();
+    /// let poly = PolyOverZ::from_coefficient_embedding(&vector);
+    /// let cmp_poly = PolyOverZ::from_str("3  17 3 -5").unwrap();
+    /// assert_eq!(cmp_poly, poly)
+    /// ```
+    ///
+    /// # Panics ...
+    /// - if the provided embedding is not a column vector.
+    fn from_coefficient_embedding(embedding: &MatZ) -> Self {
+        assert!(embedding.is_column_vector());
+        let mut out = PolyOverZ::default();
+        for i in 0..embedding.get_num_rows() {
+            out.set_coeff(i, embedding.get_entry(i, 0).unwrap())
+                .unwrap()
+        }
+        out
+    }
+}
+
+#[cfg(test)]
+mod test_into_coefficient_embedding {
+    use crate::{
+        integer::{MatZ, PolyOverZ},
+        traits::IntoCoefficientEmbedding,
+    };
+    use std::str::FromStr;
+
+    /// Ensure that the embedding works with large entries
```

Comment:
```suggestion
    /// Ensure that the embedding works with large entries.
```
Also add dots at the end of the other test descriptions.